### ISSUE TITLE
About the code of MSD

### ISSUE THREAD

**Issue by Superman-Valencia** [BlueBlaze6335/FLSGAN-VC#2]
🙂Sorry to bother you again, I still have some questions about MSD loss:
1、About getting MCEP, do you get MCEP by operating on the time dimension?
mc = tf.signal.dct(fft_mag, n=24)  # direct cosine transform (dim =24)

2、x = tf.reduce_mean(x, axis=(0,), keepdims=True)
In the above code, was the MCEP mean obtained by averaging the batch dimension?

3、How is the total dimension of MS (24) determined? Is there any correlation between 24 in the following two lines of code?
mc = tf.signal.dct(fft_mag, n=24)  # direct cosine transform (dim =24)
for i in range(24): # calculating difference across each dimension (24)

Looking forward to your reply, thank you very much！🙇‍

**Comment by BlueBlaze6335** [comment]
1. Yes
2. As of your second question, yes we obtain the mean across the batch dimension. 
3. The dim=24 is the transform dimension. It is generally kept 24 or 32 as a common practice (through literature review). There is correlation because we compute the differences across these 24 dimensions.

**Comment by BlueBlaze6335** [comment]
You can go through this [paper](https://ieeexplore.ieee.org/document/10040000).

**Comment by Superman-Valencia** [comment]
Thank you so much.
